gobcore/logging/logger.py

```diff
@@ -100,25 +106,86 @@ def __init__(self, name=None):
         if name is not None:
             self.set_name(name)
         self._default_args = {}
+        self._offload_file = None
+        self._offload_filename = None
+        self._offload_empty = True
+
         self._clear_logs()
         self.clear_issues()
 
     def clear_issues(self):
         self._issues = {}
         self._data_msg_count = defaultdict(int)
+        self._offload_file = None
+
+        # Remove the offloaded issues
+        if self._offload_filename:
+            try:
+                os.remove(self._offload_filename)
+            except OSError:
+                pass
+
+    def open_offload_file(self):
+        self._offload_filename = get_filename(get_unique_name(), self._ISSUES_FOLDER)
+        self._offload_file = open(self._offload_filename, 'w+')
+
+        self._offload_empty = True
+
+    def close_offload_file(self):
+        if self._offload_file:
+            self._offload_file.close()
 
     def add_issue(self, issue, level):
+        if not self._offload_file:
+            self.open_offload_file()
+
         id = issue.get_unique_id()
         if self._issues.get(id):
+            from gobcore.quality.issue import Issue
+
+            # Get the pointer to the issue in the offload file, seek the line and remove any comma or newline
+            issue_offset = self._issues.get(id)
+            self._offload_file.seek(issue_offset)
+
+            existing_issue_string = self._offload_file.readline().rstrip("\n")
+
             # Join this issue with an already existing issue for the same check, attribute and entity
-            self._issues[id].join_issue(issue)
+            existing_issue = Issue.from_json(existing_issue_string)
+            existing_issue.join_issue(issue)
+
+            # Return to the end of the file and write the new issue
+            self._offload_file.seek(0, io.SEEK_END)
+            self._issues[id] = self.write_issue(existing_issue)
         else:
-            # Add this issue as a new issue
-            self._issues[id] = issue
+            # Write this issue and save the byte offset to the issue
+            self._issues[id] = self.write_issue(issue)
             self._data_msg_count['data_' + level] += 1
 
+    def write_issue(self, issue):
+        if not self._offload_empty:
+            self._offload_file.write("\n")
+
+        # Store the position in the file before the issue and return it to store a pointer
+        issue_offset = self._offload_file.tell()
+
+        self._offload_file.write(issue.json)
+        self._offload_empty = False
```

Comment:
Why not always write _issue + "\n"_? Removes the _offload_empty variable and check.

Reply:
Ah true, a remainder of when I was first writing the file to be able to be parsed by ijson. Will fix